packages/slice-machine/src/domain/customType.ts

```diff
@@ -425,6 +447,41 @@ export function reorderField(args: ReorderFieldArgs): CustomType {
   return newCustomType;
 }
 
+// TODO add tests
+export function updateUidField(
+  label: string,
+  customType: CustomType,
+): CustomType {
+  const field = getUidField(customType);
+  if (!field) return customType;
+
+  const section = findSectionWithUidField(customType);
+  const sectionId = section ?? "Main";
+
+  const newField = {
+    ...field,
+    config: {
+      ...field.config,
+      label,
+    },
+  };
+
+  const updatedSection = updateFields({
+    fields: customType.json[sectionId],
+    previousFieldId: "uid",
+    newFieldId: "uid",
```

Comment:
⚠️ #issue

Today technically you CAN have a UID that is not named uid. If you create a Page Type then open the JSON file and manually change the API ID, you will see that it's possible. 
You can see that prismic-types-internal that control what can be pushed or not never enforce that it should be name UID: https://github.com/prismicio/prismic-types-internal/pull/27

The goal would be use the field id instead of hardcoding it. Since you're finding it with the type, you should have it after.

Reply:
didn't know that, will fix it